pulsar-metadata/src/test/java/org/apache/pulsar/metadata/ZKSessionTest.java

```diff
@@ -183,9 +182,10 @@ public void testReacquireLeadershipAfterSessionLost() throws Exception {
         e = sessionEvents.poll(10, TimeUnit.SECONDS);
         assertEquals(e, SessionEvent.SessionReestablished);
 
-        Awaitility.await().untilAsserted(() -> {
-                    assertEquals(le1.getState(), LeaderElectionState.Leading);
-        });
+        // due to zookeeper async notice, we need to wait.
+        Awaitility.await()
+                .until(() -> le1.getState() == LeaderElectionState.Leading);
+        assertEquals(le1.getState(), LeaderElectionState.Leading);
```

Comment:
this line is redundant because we are waiting for this condition in the Awaitility block

Reply:
Ok, I will fix it, thanks to you ~ :)

Reply:
@mattisonchao It's a good practice to use `untilAsserted` + asserts (instead of `.until`) because of the reason what Enrico explained.